openmetadata-ui/src/main/resources/ui/src/components/Assets/AssetsSelectionModal/AssetSelectionModal.tsx

```diff
@@ -266,27 +321,34 @@ export const AssetSelectionModal = ({
         .filter(Boolean);
 
       await Promise.all(patchAPIPromises);
+      await new Promise((resolve) =>
+        delay(() => resolve(''), TIMEOUT_DURATION)
+      );
       onSave?.();
-      onCancel();
     } catch (_) {
       // Nothing here
     } finally {
-      setIsLoading(false);
+      setIsSaveLoading(false);
+      onCancel();
     }
   };
 
-  const onSaveAction = () => {
+  const onSaveAction = useCallback(() => {
     if (type === AssetsOfEntity.GLOSSARY) {
       handleSave();
     } else {
       domainAndDataProductsSave();
     }
-  };
+  }, [type, handleSave, domainAndDataProductsSave]);
 
   const onScroll: UIEventHandler<HTMLElement> = useCallback(
     (e) => {
+      const scrollHeight =
+        e.currentTarget.scrollHeight - e.currentTarget.scrollTop;
+
       if (
-        e.currentTarget.scrollHeight - e.currentTarget.scrollTop === 500 &&
+        scrollHeight > 499 &&
+        scrollHeight < 501 &&
```

Comment:
Is it coming in decimals?

Reply:
@Ashish8689 can you reply here ?

Reply:
yes, @chirag-madlani  it's coming decimal, and sometimes the exact number in different screen sizes, in the above range!